fe/fe-core/src/main/java/org/apache/doris/datasource/hudi/source/HudiScanNode.java

```diff
@@ -193,6 +202,26 @@ protected void doInitialize() throws UserException {
         } else {
             incrementalRelation = null;
         }
+
+        timeline = hudiClient.getCommitsAndCompactionTimeline().filterCompletedInstants();
+        if (desc.getRef().getTableSnapshot() != null) {
+            queryInstant = desc.getRef().getTableSnapshot().getTime();
+            snapshotTimestamp = Option.of(queryInstant);
+        } else {
+            Option<HoodieInstant> snapshotInstant = timeline.lastInstant();
+            if (!snapshotInstant.isPresent()) {
+                prunedPartitions = Collections.emptyList();
+                prunedPartitionsIter = prunedPartitions.iterator();
+                return;
+            }
+            queryInstant = snapshotInstant.get().getTimestamp();
+            snapshotTimestamp = Option.empty();
+        }
+        // Non partition table will get one dummy partition
+        prunedPartitions = HiveMetaStoreClientHelper.ugiDoAs(
+                HiveMetaStoreClientHelper.getConfiguration(hmsTable),
+                () -> getPrunedPartitions(hudiClient, snapshotTimestamp));
+        prunedPartitionsIter = prunedPartitions.iterator();
```

Comment:
Before, we do partition prune in `doFinalized()` method, now you move it to `doInitialized()` method.
Are you sure we can do partition prune when initializing?